optuna/samplers/_tpe/sampler.py

```diff
@@ -389,17 +389,20 @@ def _sample_from_gmm(
         active = np.argmax(self._rng.multinomial(1, weights, size=size), axis=-1)
         trunc_low = (low - mus[active]) / sigmas[active]
         trunc_high = (high - mus[active]) / sigmas[active]
-        while True:
-            samples = truncnorm.rvs(
-                trunc_low,
-                trunc_high,
-                size=size,
-                loc=mus[active],
-                scale=sigmas[active],
-                random_state=self._rng,
+        samples = np.ones(size) * (high + 1.0)
```

Comment:
You can reduce memory allocations by using `numpy.full`.
```suggestion
        samples = np.full(size, fill_value=high + 1.0, dtype=np.float64)
```

Reply:
Thank you for the suggestion. I tested several implementations given this comment. The difference between them was small, but the suggested code was consistently faster (up to 8% in my environment) than the original code. The difference between the suggestion and other variants was negligible. So I'd like to use the suggested implementation.